spec/safe_statements_spec.rb

```diff
@@ -3484,6 +3496,95 @@ def up
               end
             end
 
+            it "does not wait to acquire a lock if same table in different schema is blocked" do
+              stub_const("PgHaMigrations::LOCK_TIMEOUT_SECONDS", 1)
+
+              ActiveRecord::Base.connection.execute("CREATE TABLE partman.bogus_table(pk SERIAL, i INTEGER)")
+
+              begin
+                thread = Thread.new do
+                  migration.safely_acquire_lock_for_table("partman.bogus_table") { sleep 2 }
```

Comment:
I wonder if we should do an explicit lock here rather than relying on our code (since our own code is also what's under test).

Reply:
Addressed in https://github.com/braintree/pg_ha_migrations/pull/87/commits/bd2c1e92e2cfd7765f579f6baec38bf66f4a1a22